### ISSUE TITLE
Need a test that exercises the packaged files

### ISSUE THREAD

**Issue by maribethb** [google/blockly#6059]
**Is your feature request related to a problem? Please describe.**

Our current modes of testing do not exercise the packaged files, i.e. those in `/dist` that are created from `npm run package`.
Some of the demos in blockly-samples do test the code in `dist/` (e.g. mirror demo), but those in core do not (e.g. the mirror demo in core references `msg/js/en.js` but NOT `dist/msg/en.js`. So there is not an easy way to test the packaged code before publishing.

**Describe the solution you'd like**

A test page inside of core that just loads Blockly from the package in `dist/`. This page can then be opened to easily smoke test the package output.

Note that most of the demos have relocated to blockly-samples, and we plan to remove them from core. So just changing the current demos is not a good/long-term solution imo.

**Describe alternatives you've considered**

This can be tested with blockly-samples, but the plugins are usually what we test and they don't exercise the code paths in the same way (they use the code as modules, but we also need to ensure the code can be included with script tags). The examples/demos are not tested as frequently, and this method requires either a) already having published a beta release of blockly or b) manually linking blockly with npm link (which you'd have to do separately for each plugin or example you want to test)

So in other words, it's possible, but not frictionless, to test this today. But we should introduce an easier path.

**Additional context**

This came up because we found that demos in blockly-samples are broken wrt message loading, but the same demos work fine in core because they test a different code path.

**Comment by BeksOmega** [comment]
I agree that there is some friction here (evidenced by the fact that testing demos has been missed in the past), but I'm confused about where it is.

You mention:
> The examples/demos are not tested as frequently, and this method requires either a) already having published a beta release of blockly or b) manually linking blockly with npm link (which you'd have to do separately for each plugin or example you want to test)

But I don't understand how (a) is a problem. Doesn't a beta always get published before bug bash week so that external developers can test if they want to?

Does it just need to be made clearer that demos should be tested in addition to plugins?

**Comment by maribethb** [comment]
Ideally, I don't want to find out about a bug during bug bash week. If we are making a change that plausibly could affect the packaged files, I want there to be a test, or a page I can open to test my change, similar to how I do a quick check on the playground before submitting a regular feature or bug fix in core.

**Comment by maribethb** [comment]
The demos now use the correct (compiled) message files so this is a bit less important. We should still do this imo, but it can wait until after we figure out what changes we make if any to the module system, loading blockly from esm directly, and if we want to update the demos to use that, etc.
